Index: enlightenment-0.24.99.20210307.39b7896df/src/bin/e_main.c
===================================================================
--- enlightenment-0.24.99.20210307.39b7896df.orig/src/bin/e_main.c
+++ enlightenment-0.24.99.20210307.39b7896df/src/bin/e_main.c
@@ -346,6 +346,9 @@ main(int argc, char **argv)
    e_util_env_set("PANTS", "ON");
    e_util_env_set("DESKTOP", "Enlightenment");
    e_util_env_set("XDG_CURRENT_DESKTOP", "Enlightenment");
+   // For now getting Qt themes to use the gtk2 theme engine is the best way
+   // to have something that works, having nothing set leads to no icons.
+   e_util_env_set("QT_QPA_PLATFORMTHEME", "gtk2");
    if (getenv("E_ALERT_FONT_DIR"))
      e_util_env_set("E_ALERT_FONT_DIR", NULL);
    if (getenv("E_ALERT_SYSTEM_BIN"))
