--- a/config.mak.dist	2021-01-18 18:27:26.545789684 +0200
+++ b/config.mak	2021-01-18 18:28:57.901504475 +0200
@@ -13,13 +13,14 @@
 # TARGET = arm-linux-musleabihf
 # TARGET = sh2eb-linux-muslfdpic
 # ...
+TARGET = or1k-linux-musl
 
 # By default, cross compilers are installed to ./output under the top-level
 # musl-cross-make directory and can later be moved wherever you want them.
 # To install directly to a specific location, set it here. Multiple targets
 # can safely be installed in the same location. Some examples:
 
-# OUTPUT = /opt/cross
+OUTPUT = /opt/cross
 # OUTPUT = /usr/local
 
 # By default, latest supported release versions of musl and the toolchain
