--- a/media/filters/ffmpeg_demuxer.cc	2021-11-09 20:43:06.650610738 +0000
+++ b/media/filters/ffmpeg_demuxer.cc	2021-11-09 21:15:01.840768565 +0000
@@ -1594,8 +1594,8 @@
   for (const auto& stream : streams_) {
     if (!stream || stream->IsEnabled() != enabled)
       continue;
-    if (av_stream_get_first_dts(stream->av_stream()) == kInvalidPTSMarker)
-      continue;
+//    if (av_stream_get_first_dts(stream) == kInvalidPTSMarker)
+//      continue;
     if (!lowest_start_time_stream ||
         stream->start_time() < lowest_start_time_stream->start_time()) {
       lowest_start_time_stream = stream.get();
@@ -1615,8 +1615,8 @@
     if (stream->type() != DemuxerStream::VIDEO)
       continue;
 
-    if (av_stream_get_first_dts(stream->av_stream()) == kInvalidPTSMarker)
-      continue;
+//    if (av_stream_get_first_dts(stream->av_stream()) == kInvalidPTSMarker)
+//      continue;
 
     if (!stream->IsEnabled())
       continue;
