--- a/buildtools/third_party/eu-strip/build.sh	2021-11-04 22:32:13.897756886 +0200
+++ b/buildtools/third_party/eu-strip/build.sh	2021-11-04 23:52:16.589236858 +0200
@@ -12,7 +12,7 @@
 patch -p1 < ../gcc-fixes.patch
 mkdir build
 cd build
-../configure --enable-maintainer-mode
+CFLAGS="-Wno-error" ../configure --enable-maintainer-mode
 make -j40
 gcc -std=gnu99 -Wall -Wshadow -Wunused -Wextra -fgnu89-inline \
   -Wformat=2 -Werror -g -O2 -Wl,-rpath-link,libelf:libdw -Wl,--build-id=none -o eu-strip \
