--- a/libreoffice.SlackBuild	2019-08-10 17:16:16.458323738 +0300
+++ b/libreoffice.SlackBuild	2019-08-11 14:56:31.363112297 +0300
@@ -1043,6 +1043,9 @@
   armv7hl)   SLKCFLAGS="-O2 -march=armv7-a -mfpu=vfpv3-d16 -std=c++98"
              SLKLDFLAGS=""; LIBDIRSUFFIX=""
              ;;
+  aarch64)   SLKCFLAGS=""
+             SLKLDFLAGS=""; LIBDIRSUFFIX="64"
+             ;;
   *)         SLKCFLAGS=""
              SLKLDFLAGS=""; LIBDIRSUFFIX=""
              ;;
@@ -1351,6 +1354,7 @@
   --with-system-libxml \
   --with-system-openssl \
   --with-system-zlib \
+  --with-system-libpng \
   --with-x \
   --without-gssapi \
   --without-junit \
