--- Makefile.in	2023-02-22 19:52:27.000000000 +0000
+++ Makefile.in	2023-03-08 18:29:29.504699161 +0000
@@ -9,7 +9,7 @@
 
 gb_Top_MODULE_CHECK_TARGETS := slowcheck unitcheck subsequentcheck perfcheck uicheck screenshot
 
-.PHONY : check-if-root bootstrap gbuild build build-non-l10n-only build-l10n-only check clean clean-build clean-host test-install distclean distro-pack-install docs download etags fetch get-submodules id install install-gdb-printers install-strip tags debugrun help showmodules translations packageinfo coverage internal.clean $(gb_Top_MODULE_CHECK_TARGETS)
+.PHONY : bootstrap gbuild build build-non-l10n-only build-l10n-only check clean clean-build clean-host test-install distclean distro-pack-install docs download etags fetch get-submodules id install install-gdb-printers install-strip tags debugrun help showmodules translations packageinfo coverage internal.clean $(gb_Top_MODULE_CHECK_TARGETS)
 
 MAKECMDGOALS?=build
 build_goal:=$(if $(filter build uicheck,$(MAKECMDGOALS)),build)\
@@ -59,18 +59,6 @@ else # MAKE_RESTARTS
 
 .DEFAULT_GOAL := build
 
-# Nested test is to check if the systemd-detect-virt command doesn't exist or it reports we're not
-# in a container:
-check-if-root:
-	@if test ! `uname` = 'Haiku' -a `id -u` = 0; then \
-		if test ! -x "$$(command -v systemd-detect-virt)" || ! systemd-detect-virt -c -q; then \
-			echo; \
-			echo 'Building LibreOffice as root is a very bad idea, use a regular user.'; \
-			echo; \
-			exit 1; \
-		fi \
-	fi
-
 gb_Side ?= host
 
 include $(BUILDDIR)/config_$(gb_Side).mk
@@ -276,7 +264,7 @@ endif
 #
 # Bootstrap
 #
-bootstrap: check-if-root compilerplugins
+bootstrap: compilerplugins
 
 #
 # Build
